### ISSUE TITLE
How to use it?

### ISSUE THREAD

**Issue by kamillopl** [JDRIVO/plugin.video.gdrive#6]
No info about how to make it work. We need to create new project and add Google Drive v3 API, then make test user and provide his credentials, it is it? Does that require every 7 days credentials update?

**Comment by JDRIVO** [comment]
You will need a project with Google Drive API privileges. From there, you will need to create a OAuth client ID in the form of a web application with this redirect URI: http://localhost:8011/status

To get started:
https://console.cloud.google.com/apis/library/drive.googleapis.com

**Comment by JDRIVO** [comment]
Credentials don't have to be updated as far as I'm aware.

**Comment by JDRIVO** [comment]
This video may be of help: https://www.youtube.com/watch?v=RwtUBOijAx4

**Comment by Mfstick** [comment]
@JDRIVO 
I tried the above mentioned steps and created OAuth client ID in the form of a web application with this redirect URI: http://localhost:8011/status

But it is showing error as Error 400: redirect_uri_mismatch
![image](https://github.com/JDRIVO/gDrive/assets/162947426/a71fc44a-abaa-4db9-8a6c-f292a9c03a8b)

**Comment by JDRIVO** [comment]
Sorry, 8011 is the default port. If you have changed the port within the addon settings (under the streamer category) then you will need change the port in the url to that. So for example, if you changed the port to 8564 within the addon then the url needs to be: http://localhost:8564/status

**Comment by JDRIVO** [comment]
"Enroll account"... I recall removing that terminology in recent versions of gDrive. And have just double checked to ensure it isn't present in the current version which leads me to believe you must be using an old version. Did you download gDrive from one of the zips listed here https://github.com/JDRIVO/gDrive/releases/tag/v1.0.0? If you did, they are obsolete. You will need to download it from https://github.com/JDRIVO/gDrive/archive/refs/heads/master.zip

**Comment by Mfstick** [comment]
@JDRIVO 
I can able to connect now after installing master.zip. But when i try to access it is showing me folder name and I am not able to see the files inside the folder.
Any other permission required to access the file.
I have already enabled google drive API before creating the token.

**Note**
Found one challenge while connecting. Add on shows URL as localhost instead of local IP for connection. It will work only when device and browser are same. Not working when we use add on and browser on different device even though they are sharing same WiFi.
We need to manually identify the local IP of the device to access the URL on another device.
In old Add on it shows local IP.

**Comment by JDRIVO** [comment]
If you hold down the select button when highlighting a folder in your google drive account, a context menu should appear presenting the option to "Sync folder". Click that. I'd recommend increasing the threads to 10+ in the sync settings category to decrease the time it takes to sync the folder.

Yes, unfortunately google made API changes so that the only way to access the google auth page is through localhost. Meaning you have to access localhost using a browser on the same device that Kodi is running. I added an import account option in case you have a secondary device that doesn't have browser access.

**Comment by Nickytun** [comment]
That's great. Thanks so much for great addon. I hope you'll continue this project and frenquency update this great addon.

**Comment by Nickytun** [comment]
Hi bro! after 7 days. I have to log in and update credentials. Does any way can't auto update it?

**Comment by JDRIVO** [comment]
Hey, I have found the issue: A Google Cloud Platform project with an OAuth consent screen configured for an external user type and a publishing status of "Testing" is issued a refresh token expiring in 7 days. 

To resolve this you need to go to the "OAuth consent screen" section of your project and publish your app, refer to the image below.

https://console.cloud.google.com/apis/credentials/consent

![image](https://github.com/user-attachments/assets/c510ed63-9166-45c7-9f19-807aa8a8e2ef)

**Comment by Nickytun** [comment]
Thanks so much bro! I've done it!

**Comment by Nickytun** [comment]
Hi bro. I publiscd app but it won't work. After 7 days. I can't play movie from addon. I have to update again.

**Comment by JDRIVO** [comment]
Did you re-register an account after publishing? I assume that an unexpiring refresh token would be only issued to a new registration.

**Comment by Nickytun** [comment]
I do bro! I re-register when publishing. But I don't know why after 7 days. I can't play. Let me try again and will conform with u. Thanks

**Comment by JDRIVO** [comment]
OK.

**Comment by Nickytun** [comment]
That's great bro! Now I used your add on without register after 7 days. Perfect! Thanks so much.

**Comment by JDRIVO** [comment]
@Nickytun  Thanks for confirming. I will have to update the tutorial video.

**Comment by JDRIVO** [comment]
You need to long press on a folder to display a context menu which will allow you to sync the folder (create strm files).

**Comment by fusionhost760** [comment]
thanks it working now

**Comment by JDRIVO** [comment]
Glad to hear.
